python-avd/pyavd/_eos_designs/structured_config/underlay/ethernet_interfaces.py

```diff
@@ -33,48 +32,44 @@ def ethernet_interfaces(self: AvdStructuredConfigUnderlayProtocol) -> None:
             description = self.shared_utils.interface_descriptions.underlay_ethernet_interface(
                 InterfaceDescriptionData(
                     shared_utils=self.shared_utils,
-                    interface=link["interface"],
-                    link_type=link["type"],
-                    peer=link["peer"],
-                    peer_interface=link["peer_interface"],
+                    interface=link.interface,
+                    link_type=link.type,
+                    peer=link.peer,
+                    peer_interface=link.peer_interface,
                 ),
             )
             ethernet_interface = EosCliConfigGen.EthernetInterfacesItem(
-                name=link["interface"],
-                peer=link["peer"],
-                peer_interface=link["peer_interface"],
-                peer_type=link["peer_type"],
+                name=link.interface,
+                peer=link.peer,
+                peer_interface=link.peer_interface,
+                peer_type=link.peer_type,
                 description=description or None,
-                speed=link.get("speed", Undefined),
-                shutdown=self.inputs.shutdown_interfaces_towards_undeployed_peers and not link["peer_is_deployed"],
+                speed=link.speed,
+                shutdown=self.inputs.shutdown_interfaces_towards_undeployed_peers and not link.peer_is_deployed,
             )
 
             # L3 interface
             # Used for p2p uplinks as well as main interface for p2p-vrfs.
-            if link["type"] == "underlay_p2p":
+            if link.type == "underlay_p2p":
                 ethernet_interface._update(
                     mtu=self.shared_utils.p2p_uplinks_mtu,
                     service_profile=self.inputs.p2p_uplinks_qos_profile,
-                    ipv6_enable=link.get("ipv6_enable", Undefined),
-                    flow_tracker=self.shared_utils.new_get_flow_tracker(
-                        link.get("flow_tracking"), output_type=EosCliConfigGen.EthernetInterfacesItem.FlowTracker
-                    ),
+                    ipv6_enable=link.ipv6_enable,
+                    flow_tracker=self.shared_utils.new_get_flow_tracker(link.flow_tracking, output_type=EosCliConfigGen.EthernetInterfacesItem.FlowTracker),
```

Comment:
we should rename new_get_flow_tracker in anotehr PR

Reply:
ack